src/draw_functions.py

```diff
@@ -106,6 +106,16 @@ def particles(sample, cr, width, height, colors, offset, radius):
             height / 10 - offset_px * 2, radius)
     cr.fill()
 
+def spine(sample, cr, width, height, colors, radius):
+    set_source(cr, height, colors)
+    ls = len(sample)
+    step = width / ls
+    for i in range(ls):
+        draw_element(cr, step * i + step / 2 - sample[i] * step / 2, \
+            height / 2 - sample[i] * 25, step * sample[i], \
```

Comment:
If the goal was to make elements aligned like on screenshot, this line needs to be fixed to not make elements jump when the window has small width:
`height / 2 - sample[i] * step / 2, step * sample[i], \`

![](https://i.imgur.com/FMLIzyM.png)

Reply:
Again, if the result on screenshot is what you wanted to get, I'm fine with the name Spine. But I also suggest Vessels - because elements kinda get filled with something 😄 Up to you, whatever you like more :)

Also, I don't think this mode should be affected by mirror, but ability to rotate this mode would be great. I suggest to rotate it automatically when window height is bigger than width:
```
if height > width:
        cr.translate(width, 0)
        cr.rotate(math.pi / 2)
        width, height = height, width
```

[Screenshot](https://i.imgur.com/US4Ydbl.png)

Reply:
I did what you said, but we have a problem now:

![Screenshot from 2023-01-29 21-26-52](https://user-images.githubusercontent.com/119804311/215354020-f653d74f-b965-4552-bfa7-cfda1f614ab5.png)

I'm still not sure about this drawing mode, it creates a bunch of problems and interferes with how the rest of the app works and with what's planned for the future.

Reply:
> I did what you said, but we have a problem now:

Ah yeah, I didn't test it well. I will try to find a fix.

> I'm still not sure about this drawing mode, it creates a bunch of problems and interferes with how the rest of the app works and with what's planned for the future.

Well it doesn't follow the basic style of other modes, but I don't see critical problems currently. I think I can make it use colors properly. And disabling mirror just for this mode is not a big deal.

Reply:
I'll work on it tomorrow, then. I think I know how to fix it.